lib/flex.js

```diff
@@ -34,26 +34,26 @@ class Service {
       delete options.port;
     }
 
-    this.dataLink = dataLink;
-    this.businessLogic = businessLogic;
+    this.data = data;
+    this.functions = functions;
     this.logger = logger;
     this.moduleGenerator = moduleGenerator;
 
     const taskReceivedCallback = (task, completionCallback) => {
-      if (task.taskType === 'dataLink') {
-        return this.dataLink.process(task, this.moduleGenerator.generate(task), completionCallback);
+      if (task.taskType === 'data') {
+        return this.data.process(task, this.moduleGenerator.generate(task), completionCallback);
       } else if (task.taskType === 'serviceDiscovery') {
         task.discoveryObjects = {
-          dataLink: {
-            serviceObjects: this.dataLink.getServiceObjects()
+          data: {
+            serviceObjects: this.data.getServiceObjects()
           },
-          businessLogic: {
-            handlers: this.businessLogic.getHandlers()
+          functions: {
+            handlers: this.functions.getHandlers()
           }
         };
         return completionCallback(null, task);
-      } else if (task.taskType === 'businessLogic') {
-        return this.businessLogic.process(task, this.moduleGenerator.generate(task), completionCallback);
+      } else if (task.taskType === 'functions') {
```

Comment:
Ah, it's also in KCS and MAPI. Ok, will revert to allow both for now but I think that it's good to have the internals match the reality as much as possible to avoid confusion..